projects/experimental/directives/pin/pin.directive.ts

```diff
@@ -0,0 +1,54 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Directive,
+    EventEmitter,
+    HostListener,
+    Inject,
+    Input,
+    Output,
+    ViewEncapsulation,
+} from '@angular/core';
+import {TUI_PLATFORM, TuiDirectiveStylesService, TuiPlatform} from '@taiga-ui/cdk';
+
+@Component({
+    host: {class: 'tui-pin'},
+    template: '',
+    styleUrls: ['./pin.styles.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+})
+export class TuiPinStyleComponent {}
+
+@Directive({
+    selector: '[tuiPin]',
+    host: {
+        tuiPin: '',
+        '[attr.tabindex]': '0',
```

Comment:
Let's not make it focusable by default. Since it's an attribute directive, users can put it on buttons or links if they want them focusable. Let's add `clearbtn` and `text-decoration: none` for such cases.

Reply:
fixed